### PR TITLE
Ocarina of Time: Itemlinks and bugfixes

### PR DESCRIPTION
**Pull request by espeon65536**
## What is this fixing or adding?
- Songs and dungeon items can be used with itemlink with various fill settings. The linked items will only appear in locations that could contain the original items, spread between players. Song locations that don't contain songs will not contain any useful items, if songs were placed there.
- The trials hint only appears when the number of required trials was not known at the time of generation.
- Dungeon maps now tell the player if dungeons are MQ or normal if the number was randomized at the time of generation, even if it is 0 or 12.
- Internal connect name is now randomized to a 16-character string; being able to connect to the incorrect room with a ROM now has probability 1 in 100 nonillion.
- Shop junk items are now fast-filled, which slightly improves generation time and should fix some rare generation failures.
- Fixed data bleed when opening a new ROM without properly shutting down the client first.
- Fixed No Logic failing to roll with entrance randomizer.
- Fixed Deku Shield potentially not being purchasable in closed forest with interior entrances randomized.
- Fixed the remaining locations not sending immediately upon pickup in game.
- Fixed crash when placing songs or dungeon items as nonlocal when they were forced local by settings. The local items are now favored, in accordance with AP's general treatment of such conflicts.
- Fixed barren hint oversight that would cause some regions with useful items to be incorrectly marked as barren.
- Fixed Gerudo Membership Card location appearing as checked with open fortress.

### CODE REVIEW

File: worlds/oot/__init__.py [hunk 1]
```diff
@@ -595,23 +620,9 @@ def generate_basic(self):  # mostly killing locations that shouldn't exist by se
...
-            "Shadow Temple Bongo Bongo Heart",
-            "Spirit Temple Twinrova Heart",
-            "Song from Impa",
-            "Sheik in Ice Cavern",
-            # only one exists
-            "Bottom of the Well Lens of Truth Chest", "Bottom of the Well MQ Lens of Truth Chest",
-            # only one exists
-            "Gerudo Training Ground Maze Path Final Chest", "Gerudo Training Ground MQ Ice Arrows Chest",
-        ]
+        def get_names(items):
```
Comment (Berserker66) [inline worlds/oot/__init__.py:623]: duplicated get_names function? Seems this is obsolete code?

File: worlds/oot/Options.py [hunk 2]
```diff
@@ -1,9 +1,34 @@
 import typing
+import random
 from Options import Option, DefaultOnToggle, Toggle, Range, OptionList, DeathLink
 from .LogicTricks import normalized_name_tricks
 from .ColorSFXOptions import *
 
 
+class TrackRandomRange(Range):
```
Comment (Berserker66) [inline worlds/oot/Options.py:8]: A bit sad to see this be OoT specific. I think Violet wanted something similar for Witness.

File: worlds/oot/__init__.py (hunk 1 above)
Comment (espeon65536) [reply worlds/oot/__init__.py:623]: probably got accidentally duplicated in the merge. It's definitely not obsolete though

File: worlds/oot/Options.py (hunk 2 above)
Comment (espeon65536) [reply worlds/oot/Options.py:8]: I couldn't think of a good way to make this work for all option types besides overhauling Generate.py, and I'd rather do that in a separate PR. If that happens, this randomization tracking will be extended to all options that support weighting.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**